storage/connect/jdbconn.cpp

```diff
@@ -451,8 +451,14 @@ PQRYRES JDBCSrcCols(PGLOBAL g, PCSZ src, PJPARM sjp)
 
 	if (strstr(src, "%s")) {
 		// Place holder for an eventual where clause
-		sqry = (char*)PlugSubAlloc(g, NULL, strlen(src) + 2);
-		sprintf(sqry, src, "1=1");			 // dummy where clause
+		size_t sqry_size = strlen(src) + 2;
+		sqry = (char*)PlugSubAlloc(g, NULL, sqry_size);
+		// Function PlugSubAlloc(...) recalculate string size
+		// while allocate memory - it rounds size up size to multiple of 8
+		// we need to know the real allocated size (SQRY_REAL_ALLOCATED_SIZE)
+		// to use it in sprintf(...)
+		const int SQRY_REAL_ALLOCATED_SIZE = ((sqry_size + 7) / 8) * 8;
```

Comment:
The need for this comment is an indicator that this is a bit of a "leaky abstraction".

Rather than duplicate the code, what if we change the signature of PlugSubAlloc to take a `size_t *` so the caller can see how it is modified?

If that change would be too invasive, what about making  `SQRY_REAL_ALLOCATED_SIZE` a macro that both PlugSubAlloc and this code use?

Reply:
> Rather than duplicate the code, what if we change the signature of PlugSubAlloc to take a `size_t *` so the caller can see how it is modified?
> 
> If that change would be too invasive, then indeed I'd support changing `SQRY_REAL_ALLOCATED_SIZE` to a macro as @LinuxJedi suggested above, and have that macro used both in PlugSubAlloc and this code use.

Thanks for the review. Yes I think changing the signature of PlugSubAlloc() looks too invasive and can be not very obvious for other developers.
Let's stick with macro - I added ROUNDUP_TO_8 macro to get rid of code duplication